plugins/lsp/language-servers/pylsp.nix

```diff
@@ -528,21 +532,98 @@ in {
+                      filter
+                      (dep: dep.pname != "python-lsp-server")
+                      old.propagatedBuildInputs;
+
+                    # Skip testing because those naked dependencies will complain about missing pylsp
+                    doCheck = false;
+                  }
+                )
+              )
             )
-            (with pkgs.python3Packages; {
-              pylsp_mypy = pylsp-mypy;
-              isort = pyls-isort;
-              black = python-lsp-black;
-              memestra = pyls-memestra;
-              rope = pylsp-rope;
-              ruff = python-lsp-ruff;
-            })
+          )
+          (with pkgs.python3Packages; {
+            pylsp_mypy = pylsp-mypy.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  substituteInPlace setup.cfg \
+                    --replace "python-lsp-server >=1.7.0" ""
+                '';
+            });
+            isort = pyls-isort.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  substituteInPlace setup.py \
+                    --replace 'install_requires=["python-lsp-server", "isort"],' 'install_requires=["isort"],'
+                '';
+            });
+            black = python-lsp-black.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  substituteInPlace setup.cfg \
+                    --replace "python-lsp-server>=1.4.0;" ""
+                '';
+            });
+            memestra = pyls-memestra.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  sed -i '/python-lsp-server/d' requirements.txt
+                '';
+            });
+            rope = pylsp-rope.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  sed -i '/python-lsp-server/d' setup.cfg
+                '';
+            });
+            ruff = python-lsp-ruff.overridePythonAttrs (old: {
+              postPatch =
+                old.postPatch
+                or ''''
+                + ''
+                  sed -i '/python-lsp-server/d' pyproject.toml
+                '';
+
+              nativeBuildInputs = [setuptools] ++ (old.nativeBuildInputs or []);
+            });
+          })
+        );
+
+        # Final list of pylsp plugins to install
+        pylspPlugins = nativePlugins ++ thirdPartyPlugins;
+      in
+        mkDefault (
+          # This is the final default package for pylsp
+          pkgs.python3Packages.python-lsp-server.overridePythonAttrs (
+            old: {
+              propagatedBuildInputs = pylspPlugins ++ old.propagatedBuildInputs;
```

Comment:
I am not sure rather we should add `doCheck = false;` here.
In practice, the testing is annoying for the user and lasts for ~30s.
On the other hand, keeping it like this (enabled) adds extra robustness as it would cause our CI to fail in case of a problem.

What do you think ?

Reply:
Honestly, I'd say `doCheck = true`, especially since we can get CI up and running now. But we can probably add an option that allows disabling that.